src/main/java/uk/gov/companieshouse/api/strikeoffobjections/file/FileTransferApiClient.java

```diff
@@ -167,15 +175,84 @@ public FileTransferApiClientResponse delete(String requestId, String fileId) {
-        FileTransferApiClientResponse fileTransferApiClientResponse = new FileTransferApiClientResponse();
-        fileTransferApiClientResponse.setFileId(fileId);
-        fileTransferApiClientResponse.setHttpStatus(HttpStatus.OK);
+    public FileTransferApiClientResponse download(String requestId, String fileId, HttpServletResponse httpServletResponse) {
+        String downloadUri = String.format(DOWNLOAD_URI, fileTransferApiURL, fileId);
+
+        return makeApiCall(
+                requestId,
+                //FileTransferOperation
+                () -> restTemplate.execute(
+                        downloadUri,
+                        HttpMethod.GET,
+                        requestCallback -> {
+                            handleRequestCallback(requestCallback);
+                        },
+                        clientHttpResponse -> {
+                            return getClientHttpResponse(httpServletResponse, clientHttpResponse);
+                        }),
+
+                //FileTransferResponseBuilder - the output from FileTransferOperation is the input into
+                // this FileTransferResponseBuilder
+                clientHttpResponse -> {
+                    return getFileTransferApiClientResponse(clientHttpResponse);
+                }
+        );
+    }
+
+    private boolean isFileClean(String requestId, String fileId) {
+        String getUrl = String.format(GET_URI, fileTransferApiURL, fileId);
+
+        FileTransferApiClientResponse fileDataResponse = makeApiCall(
+                requestId,
+                () -> {
+                    HttpEntity<Void> request = new HttpEntity<>(createApiKeyHeader());
+                    return restTemplate.exchange(getUrl, HttpMethod.DELETE, request, FileTransferApiResponse.class);
+                },
+                responseEntity -> {
+                    return getFileTransferApiClientResponse(requestId, responseEntity);
+                }
+        );
+
+        return fileDataResponse.getAvStatus().equals("clean");
+
+    }
+
+    private void handleRequestCallback(ClientHttpRequest requestCallback) {
+        requestCallback.getHeaders().setAccept(Collections.singletonList(MediaType.APPLICATION_OCTET_STREAM));
+        requestCallback.getHeaders().add(HEADER_API_KEY, fileTransferApiKey);
+    }
+
+    private ClientHttpResponse getClientHttpResponse(HttpServletResponse httpServletResponse, ClientHttpResponse clientHttpResponse) throws IOException {
+        setResponseHeaders(httpServletResponse, clientHttpResponse);
 
-        InputStream inputStream = new ByteArrayInputStream("This is a test".getBytes());
+        InputStream inputStream = clientHttpResponse.getBody();
         IOUtils.copy(inputStream, httpServletResponse.getOutputStream());
+        return clientHttpResponse;
+    }
 
+    private FileTransferApiClientResponse getFileTransferApiClientResponse(ClientHttpResponse clientHttpResponse) throws IOException {
```

Comment:
Not sure, but do you need to declare the 'throws IOException' on this method. Try removing it and see if code compiles.

Reply:
No, won't compile without them